server_addon/substancepainter/server/settings/load_plugins.py

```diff
@@ -0,0 +1,120 @@
+        {"label": "2048", "value": 2048},
+        {"label": "4096", "value": 4096}
+    ]
+
+
+class ProjectTemplatesModel(BaseSettingsModel):
+    _layout = "expanded"
+    name: str = SettingsField("default", title="Template Name")
+    default_texture_resolution: int = SettingsField(
+        1024, enum_resolver=document_resolution_enum,
+        title="Document Resolution",
+        description=("Set texture resolution when "
+                     "creating new project.")
+    )
+    import_cameras: bool = SettingsField(
+        True, title="Import Cameras",
+        description="Import cameras from the mesh file.")
+    normal_map_format: str = SettingsField(
+        "DirectX", enum_resolver=normal_map_format_enum,
+        title="Normal Map Format",
+        description=("Set normal map format when "
+                     "creating new project.")
+    )
+    project_workflow: str = SettingsField(
+        "Default", enum_resolver=uv_workflow_enum,
+        title="UV Tile Settings",
+        description=("Set UV workflow when "
+                     "creating new project.")
+    )
+    tangent_space_mode: str = SettingsField(
+        "PerFragment", enum_resolver=tangent_space_enum,
+        title="Tangent Space",
+        description=("An option to compute tangent space "
+                     "when creating new project.")
+    )
+    preserve_strokes: bool = SettingsField(
+        True, title="Preserve Strokes",
+        description=("Preserve strokes positions on mesh.\n"
+                     "(only relevant when loading into "
+                     "existing project)")
+    )
+
+
+class ProjectTemplateSettingModel(BaseSettingsModel):
+    project_templates:  list[ProjectTemplatesModel] = SettingsField(
+        default_factory=ProjectTemplatesModel,
+        title="Project Templates"
+)
+
+
+class LoadersModel(BaseSettingsModel):
+    SubstanceLoadProjectMesh: ProjectTemplateSettingModel = SettingsField(
+        default_factory=ProjectTemplateSettingModel,
+        title="Load Mesh"
+    )
+
+
+DEFAULT_LOADER_SETTINGS = {
+    "SubstanceLoadProjectMesh":{
+        "project_templates": [{
+            "name": "2K(Default)",
+            "default_texture_resolution": 2048,
+            "import_cameras": True,
+            "normal_map_format": "DirectX",
+            "project_workflow": "Default",
+            "tangent_space_mode": "PerFragment",
+            "preserve_strokes": True
+        },
+        {
+            "name": "2K(UV tile)",
+            "default_texture_resolution": 2048,
+            "import_cameras": True,
+            "normal_map_format": "DirectX",
+            "project_workflow": "UVTile",
+            "tangent_space_mode": "PerFragment",
+            "preserve_strokes": True
+        },
+        {
+            "name": "4K(Custom)",
+            "default_texture_resolution": 4096,
+            "import_cameras": True,
+            "normal_map_format": "OpenGL",
+            "project_workflow": "UVTile",
```

Comment:
Shouldn't the project templates settings key be a dict instead of list so that the `name` is ensured to be unique?

Reply:
It supports multiple template settings. so that's why it is a list of the dicts

Reply:
Sorry, how does it support multiple template settings? It support multiple, by name, right?

So in typing it would be:
```python
project_templates: Dict[str, ProjectTemplatesModel]
```

So that you have e.g.

```
{
    "2K(Default)": {... default template ...},
    "2K(UV tile)": {.. other template..}
    "4K(Custom)": { .. custom .. }
}
```

Also, as default I'd have NO idea what 4K(Custom) means - as such, it's just an odd default?

Additionally I'd argue that `preserve_strokes` should really be taken out of the templates, that's really an artist decision on updates. `import_cameras` I'd say is as well. It doesn't usually influence the exports, and it's up to the artists whether cameras could help their workflow. Right?

Reply:
Hmm, doesn't seem like settings support `Dict` like that @martastain @iLLiCiTiT , correct?
It usually seems hacked [like this](https://github.com/ynput/ayon-core/blob/3d046fc2161929c46e514a71351e7cc528ff1861/server_addon/maya/server/settings/publishers.py#L249-L257).

Is there maybe a way for the backend to 'automatically' convert `Dict[str, value]` to `List[value]` with a unique name validator if that's what is apparently needed to support `dict` like structures? That would at least make defining dict like settings way easier?